### PR TITLE
Bratseth/dont remove recall neutral terms

### PR DESCRIPTION
**Pull request by bratseth**
@baldersheim please review

The semantic change is in the last commit.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/TenantController.java [hunk 1]
```diff
@@ -197,6 +197,7 @@ public Tenant migrateTenantToAthens(TenantId tenantId,
                                         PropertyId propertyId,
                                         Property property,
                                         NToken nToken) {
+        if (1==1) throw new RuntimeException("hei");
```
Comment (bjorncs) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/TenantController.java]: Looks pretty functional to me :)

File: container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java [hunk 2]
```diff
@@ -156,53 +150,45 @@ private static Recall optimizeNotItemByRestrict(NotItem item, String restrictPar
...
-                    return Recall.RECALLS_NOTHING;
-                } else if (item instanceof RankItem) {
-                    item.removeItem(i);
-                } else {
-                    throw new UnsupportedOperationException(item.getClass().getName());
-                }
-                break;
+                case RECALLS_EVERYTHING:
+                    if ((item instanceof OrItem) || (item instanceof EquivItem)) {
+                        recall = Recall.RECALLS_EVERYTHING;
```
Comment (baldersheim) [inline container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java]: Should you not check if term or field is of type filter ?

Comment (bratseth) [reply container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java]: Do you mean, keep removing them if isRanked is true?

Comment (baldersheim) [reply container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java]: Correct.

Comment (bratseth) [reply container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java]: Ok, done.

**Pull request merged**